### PR TITLE
⭐️ add aws cloudfront resource: distributions and functions

### PR DESCRIPTION
**Pull request by vjeffrey**
per user request in slack, we hadn't yet exposed the aws cloud front resource.

```
cnquery> aws.cloudfront { distributions { *} functions { * } }
aws.cloudfront: {
  distributions: [
    0: {
      origins: [
        0: aws.cloudfront.distribution.origin id = 185972265011/test-1be01d1424077260.elb.us-east-1.amazonaws.com
      ]
      status: "Deployed"
      cacheBehaviors: []
      domainName: "d1w4eig1i8et92.cloudfront.net"
      arn: "arn:aws:cloudfront::185972265011:distribution/E3J92HBG5Z8S6Q"
      defaultCacheBehavior: {
        AllowedMethods: {
          CachedMethods: {
            Items: [
              0: "HEAD"
              1: "GET"
            ]
            Quantity: 2.000000
          }
          Items: [
            0: "HEAD"
            1: "GET"
          ]
          Quantity: 2.000000
        }
        CachePolicyId: "658327ea-f89d-4fab-a63d-7e88639e58f6"
        Compress: true
        DefaultTTL: null
        FieldLevelEncryptionId: ""
        ForwardedValues: null
        FunctionAssociations: {
          Items: null
          Quantity: 0.000000
        }
        LambdaFunctionAssociations: {
          Items: null
          Quantity: 0.000000
        }
        MaxTTL: null
        MinTTL: null
        OriginRequestPolicyId: null
        RealtimeLogConfigArn: null
        ResponseHeadersPolicyId: null
        SmoothStreaming: false
        TargetOriginId: "test-1be01d1424077260.elb.us-east-1.amazonaws.com"
        TrustedKeyGroups: {
          Enabled: false
          Items: null
          Quantity: 0.000000
        }
        TrustedSigners: {
          Enabled: false
          Items: null
          Quantity: 0.000000
        }
        ViewerProtocolPolicy: "allow-all"
      }
    }
  ]
  functions: [
    0: {
      status: ""
      arn: "arn:aws:cloudfront:global:185972265011::/functions/vjtest"
      comment: ""
      stage: "DEVELOPMENT"
      name: "vjtest"
      runtime: "cloudfront-js-1.0"
      lastModifiedTime: "2023-01-29T21:07:01Z"
      createdTime: "2023-01-29T21:07:01Z"
    }
  ]
}
cnquery> exit
```

### CODE REVIEW

File: resources/packs/aws/aws_cloudfront.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	var marker *string
+	for {
+		distributions, err := svc.ListDistributions(ctx, &cloudfront.ListDistributionsInput{Marker: marker})
+		if err != nil {
+			return nil, errors.Wrap(err, "could not gather aws cloudfront distributions")
+		}
+
+		for i := range distributions.DistributionList.Items {
+			d := distributions.DistributionList.Items[i]
+			origins, err := core.JsonToDict(d.Origins)
```
Comment (imilchev) [inline resources/packs/aws/aws_cloudfront.go]: In GCP for such dicts I used manual mapping of the API object to JSON. If we just convert the API object directly to dict, we risk breaking the MQL resource when updating the AWS SDK library. Perhaps the question is do we treat the dict fields as stable MQL resources or not?

Comment (vjeffrey) [reply resources/packs/aws/aws_cloudfront.go]: that's a good point. i'll switch this up

Comment (vjeffrey) [reply resources/packs/aws/aws_cloudfront.go]: i ended up converting some of the former dict fields into fields on the resource or a list of defined resources. left the cache behaviors as a straight object -> dict translation, the caching info for thecloudfront distribution

File: resources/packs/aws/aws.lr [hunk 2]
```diff
@@ -664,6 +664,64 @@ aws.cloudtrail {
...
+  // Name of the CloudFront function
+  name string
+  // Status of the CloudFront function
+  status string
+  // ARN of the CloudFront function
+  arn string
+  // Date and time when function was last updated
+  lastModifiedTime string
+  // Date and time when function was created
+  createdTime string
```
Comment (preslavgerchev) [inline resources/packs/aws/aws.lr]: Isn't it better if we use the `time` type for time-related properties?

Comment (vjeffrey) [reply resources/packs/aws/aws.lr]: yupp, i'll change that

**Review by preslavgerchev — APPROVED** [review]

**Comment by tas50** [comment]
@vjeffrey Can you make sure to get the descriptions into the .lr files so the docs ship with the descriptions


```
+| [aws.cloudfront](aws.cloudfront.md)                                                 | AWS CloudFront                                                                                                   |
+| [aws.cloudfront.distribution](aws.cloudfront.distribution.md)                       |                                                                                                                  |
+| [aws.cloudfront.distribution.origin](aws.cloudfront.distribution.origin.md)         |
```

**Comment by vjeffrey** [comment]
ya! wanna remind me what file that's in/what file diff you're showing there?

**Comment by tas50** [comment]
The diff is from having `docs` and `cnquery` repos checked out in the same base dir and then running `make lr/docs/markdown` in the cnquery dir. That'll generate MQL docs. Run `make fmt` in the docs dir and then you'll get a clean diff of just changes.

**Comment by tas50** [comment]
https://github.com/mondoohq/cnquery/blob/main/resources/packs/aws/aws.lr#L754 is where things are missing.

**Pull request merged**